app/models/duffel.rb

```diff
@@ -7,6 +7,39 @@ def initialize(contents)
 
   def add_item(item_id)
     contents[item_id.to_s] ||= 0
-    contents[item_id.to_s] += 0
+    contents[item_id.to_s] += 1
+  end
+
+  def total
+    contents.values.sum
+  end
+
+  def count_of(item_id)
+    contents[item_id.to_s]
+  end
+
+  def subtotal
+    item_prices = contents.map do |item_id, quantity|
+      item = Item.find(item_id)
+      item.price * quantity
+    end
+    item_prices.sum
+  end
+
+  def duffel_items
+    contents.map do |item_id, _quantity|
+      Item.find(item_id)
+    end
+  end
+
+  def item_details
```

Comment:
Close, actually this is much more than I meant. We need access to all those values but most of it is already accessible in the view since we're going to iterate through items there. The only values not accessible directly via item are 'quantity' and 'item_subtotal'. We have #count_of which let's us call @duffle.count_of(item.id) to get the 'quantity', so we just need a method for the subtotal like #item_subtotal equaling what you set subtotal to. Sorry I didn't explain that well originally.